### PR TITLE
resolve: Adjust and unify D-Bus call timeout

### PR DESCRIPTION
**Pull request by ott**
DNS queries have a timeout of DNS_TRANSACTION_ATTEMPTS_MAX *
DNS_TIMEOUT_MAX_USEC = 120 s. Calls to the ResolveHostname method of
the org.freedesktop.resolve1.Manager interface have various call
timeouts that are smaller than 120 s. So it seems correct to adjust
the call timeout to the maximum query timeout and to unify the call
timeout among all callers.

A timeout of 120 s might seem large, in particular since BIND does seem
to have a query timeout of 10 s. However, it seems match the timeout
value of 120 s of Unbound. Moreover, the query and timeout handling of
resolve have problems and might be improved in the future, so this
change is at best an interin solution.

### CODE REVIEW

**Comment by poettering** [comment]
hmm, any chance you can unify this under one macro that is used at all place, to make this easier to fiddle with in the future? resolved-def.h appears like an OK place to define this in. (though it might need to be moved to src/shared/ or so, so that nss-systemd can easily inlcude it too)

**Comment by ott** [comment]
I have created a macro for the query timeout.

File: src/resolve/resolved-def.h [hunk 1]
```diff
@@ -22,18 +22,22 @@
...
+#define SD_RESOLVED_NO_CNAME           (UINT64_C(1) << 5)
+#define SD_RESOLVED_NO_TXT             (UINT64_C(1) << 6)
+#define SD_RESOLVED_NO_ADDRESS         (UINT64_C(1) << 7)
+#define SD_RESOLVED_NO_SEARCH          (UINT64_C(1) << 8)
+#define SD_RESOLVED_AUTHENTICATED      (UINT64_C(1) << 9)
+
+#define SD_RESOLVED_LLMNR              (SD_RESOLVED_LLMNR_IPV4|SD_RESOLVED_LLMNR_IPV6)
+#define SD_RESOLVED_MDNS               (SD_RESOLVED_MDNS_IPV4|SD_RESOLVED_MDNS_IPV6)
+
+#define SD_RESOLVED_PROTOCOLS_ALL      (SD_RESOLVED_MDNS|SD_RESOLVED_LLMNR|SD_RESOLVED_DNS)
```
Comment (poettering) [inline src/resolve/resolved-def.h]: uh, no need to reindent these definitions... I mean, aligning similar stuff whith whitespace is one thing, but the flags definitions defined here have little to do with the timeout definition you are adding.

In the interest of keeping patches short, can you leave this part as is, and just add the new time definition?

File: src/resolve/test-dnssec-complex.c [hunk 2]
```diff
@@ -27,10 +27,11 @@
 #include "bus-common-errors.h"
 #include "dns-type.h"
 #include "random-util.h"
+#include "resolved-def.h"
 #include "string-util.h"
 #include "time-util.h"
 
-#define DNS_CALL_TIMEOUT_USEC (45*USEC_PER_SEC)
+#define DNS_CALL_TIMEOUT_USEC SD_RESOLVED_QUERY_TIMEOUT_USEC
```
Comment (poettering) [inline src/resolve/test-dnssec-complex.c]: please drop DNS_CALL_TIMEOUT_USEC complete, and just use the new macro where we used it before

File: src/resolve/resolve-tool.c [hunk 3]
```diff
@@ -41,7 +41,7 @@
 #include "strv.h"
 #include "terminal-util.h"
 
-#define DNS_CALL_TIMEOUT_USEC (90*USEC_PER_SEC)
+#define DNS_CALL_TIMEOUT_USEC SD_RESOLVED_QUERY_TIMEOUT_USEC
```
Comment (poettering) [inline src/resolve/resolve-tool.c]: please drop DNS_CALL_TIMEOUT_USEC and use the new macro wherever it was used before

File: src/nss-resolve/nss-resolve.c [hunk 4]
```diff
@@ -30,14 +30,15 @@
...
+#include "resolved-def.h"
 #include "string-util.h"
 #include "util.h"
 #include "signal-util.h"
 
 NSS_GETHOSTBYNAME_PROTOTYPES(resolve);
 NSS_GETHOSTBYADDR_PROTOTYPES(resolve);
 
-#define DNS_CALL_TIMEOUT_USEC (45*USEC_PER_SEC)
+#define DNS_CALL_TIMEOUT_USEC SD_RESOLVED_QUERY_TIMEOUT_USEC
```
Comment (poettering) [inline src/nss-resolve/nss-resolve.c]: please drop DNS_CALL_TIMEOUT_USEC and use the new macro wherever it was used before

File: src/resolve/resolved-def.h [hunk 5]
```diff
@@ -22,18 +22,22 @@
...
+
+#define SD_RESOLVED_QUERY_TIMEOUT_USEC (120 * USEC_PER_SEC)
```
Comment (poettering) [inline src/resolve/resolved-def.h]: hmm, could you add a brief comment here, explaining why 120s is chosen here? i.e. the short version of what you wrote in the commit msg (i.e. that this is what the server side uses which is also what Unbound uses)

**Comment by ott** [comment]
I made the requested changes. I derived `DNS_TIMEOUT_MAX_USEC` based on `SD_RESOLVED_QUERY_TIMEOUT_USEC` and `DNS_TRANSACTION_ATTEMPTS_MAX` and therefore think that `SD_RESOLVED_QUERY_TIMEOUT_USEC` is self-explanatory.

**Comment by poettering** [comment]
lgtm

**Pull request merged**